File: src/_1ProgrammingBasicsWithJavaMay2023/_2ConditionalStatements/_3MoreExercise/_02SleepyTomCat.java
package _1ProgrammingBasicsWithJavaMay2023._2ConditionalStatements._3MoreExercise;

import java.util.Scanner;

public class _02SleepyTomCat {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int numbersOfHoliday = Integer.parseInt(scanner.nextLine());
        int year = 365;
        int norm = 30000;
        int workingDays = year - numbersOfHoliday;
        int playInWorkingDay = workingDays * 63;
        int playInHoliday = numbersOfHoliday * 127;
        int totalTimeToPlay = playInHoliday + playInWorkingDay;
        int diff = Math.abs(totalTimeToPlay - norm);
        int hours = diff / 60;
        int minutes = diff % 60;
        if (totalTimeToPlay > norm) {
            System.out.printf("Tom will run away%n%d hours and %d minutes more for play", hours, minutes);
        } else {
            System.out.printf("Tom sleeps well%n%d hours and %d minutes less for play", hours, minutes);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_2ConditionalStatements/_3MoreExercise/_02SleepyTomCat.js
function sleepyTomCat(holidays) {
    const tomTimeToPlayToCanSleepEnough = 30000;
    const minutesToPlayInWorkingDays = 63;
    const minutesToPlayInHolidayDays = 127
    const daysInOneYear = 365;

    const workingDays = daysInOneYear - holidays;

    const timeToPlayInHolidays = holidays * minutesToPlayInHolidayDays;
    const timeToPlayInWorkingDays = workingDays * minutesToPlayInWorkingDays;

    const totalTimeToPlay = timeToPlayInHolidays + timeToPlayInWorkingDays;

    const diff = Math.abs(tomTimeToPlayToCanSleepEnough - totalTimeToPlay);
    const hours = Math.floor(diff / 60);
    const minutes = diff % 60;

    if (totalTimeToPlay > tomTimeToPlayToCanSleepEnough) {
        console.log(`Tom will run away`);
        console.log(`${hours} hours and ${minutes} minutes more for play`);
    } else {
        console.log(`Tom sleeps well`);
        console.log(`${hours} hours and ${minutes} minutes less for play`);
    }
}

sleepyTomCat(20)
sleepyTomCat(113)